Docs: More BE/FE sweep (#2076)
grafana/plugin-tools: docusaurus/docs/key-concepts/plugin-lifecycle.md

A Grafana plugin backend component has its own separate life cycle. So long as the plugin backend is running, Grafana will make sure to restart the backend in case it crashes or is terminated. When Grafana is shut down, the backend processes is then terminated.

### Phase 6. Client-side loading

After Grafana has started and the [HTTP API](https://grafana.com/docs/grafana/latest/developers/http_api/) is running, you'll receive the server-side rendered index page containing bootstrap data. This data includes the list of available plugins and a URI to a `module.js` file that Grafana uses to instantiate the plugin.

When you interact with a UI that requires a plugin, Grafana will _lazy load_ the plugin's `module.js` file:

- For **panel plugins**, when you open a dashboard with panels (or interact with any UI that requires a plugin), Grafana lazy-loads the necessary plugin’s code through a fetch request. Each plugin is loaded only once but its objects are initialized multiple times.

> - **Data-source plugins** have multiple ways to load. For instance, if you select the data source in the dropdown, the plugin is loaded in the Explore page.  

Reviewer: ```suggestion
- **Data-source plugins** have multiple ways to load. For instance, if you select the data source in the dropdown selector in the Explore page or if you load a dashboard using it.  
```
